src/components/Pane.jsx

```diff
@@ -0,0 +1,47 @@
+var React = require('react'),
+    classNames = require('classnames'),
+    style = require('./component-playground.less');
+
+export default React.createClass({
+  displayName: 'Pane',
+
+  propTypes: {
+    split: React.PropTypes.oneOf(['vertical', 'horizontal']),
+    children: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function() {
+    return {};
+  },
+
+  render: function() {
+    var classes = {};
+    classes[style[this.props.split]] = true;
+    classes[style['pane']] = true;
+    classes = classNames(classes);
+
+    var split = this.props.split || 'vertical';
+
+    var styles = {
+      flex: 1,
+      position: 'relative',
+      outline: 'none',
+      overflow: 'none'
+    };
+    if (this.state.size) {
```

Comment:
It seems that `SplitPane` does a `setState` on this thus setting the state. This behaviour should be avoided in favour of using props. So make this component receive the size through props from its parent.